### PR TITLE
Docs to build caffe2 on ROCm

### PR DESCRIPTION
**Pull request by rohithkrn**
Initial draft of documentation to build caffe2 on ROCm inside a docker and run binary tests.

### CODE REVIEW

File: rocm-docs/caffe2-build.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# rocm-caffe2: Building From Source
```
Comment (petrex) [inline rocm-docs/caffe2-build.md]: pls drop the "rcom-caffe2" here

File: rocm-docs/caffe2-build.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+Install ROCm stack following steps at [link](https://github.com/RadeonOpenCompute/ROCm/blob/master/README.md) if your machine doesn't have ROCm already.
+
+## Build caffe2 inside docker
+
+ If your machine doesn't have docker installed, follow the steps [here](https://docs.docker.com/install/linux/docker-ce/ubuntu/#install-docker-ce) to install docker.
+
+### Pull the docker image
+```
+docker pull rohith612/caffe2:rocm1.8.2
```
Comment (petrex) [inline rocm-docs/caffe2-build.md]: Can you publish the base docker on https://hub.docker.com/r/rocm/caffe2/? thx

File: rocm-docs/caffe2-build.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+### Pull the latest caffe2 source:
+* Using https 
+```
+git clone --recursive https://github.com/pytorch/pytorch.git
+```
+* Using ssh
+```
+git clone --recursive git@github.com:pytorch/pytorch.git
+```
```
Comment (petrex) [inline rocm-docs/caffe2-build.md]: maybe we need submodule init/update here?

File: rocm-docs/caffe2-build.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+Navigate to repo directory
+```
+cd pytorch
+```
+
+### Launch the docker container
+```	
+docker run -it --network=host --device=/dev/kfd --device=/dev/dri --group-add video -v $PWD:/pytorch rohith612/caffe2:rocm1.8.2
```
Comment (petrex) [inline rocm-docs/caffe2-build.md]: pls create official docker

**Comment by petrex** [comment]
@dagamayank for feedback

File: rocm-docs/caffe2-build.md (hunk 1 above)
Comment (rohithkrn) [reply rocm-docs/caffe2-build.md]: Done.

File: rocm-docs/caffe2-build.md (hunk 2 above)
Comment (rohithkrn) [reply rocm-docs/caffe2-build.md]: Asked Peng to publish on rocm/caffe2

File: rocm-docs/caffe2-build.md (hunk 3 above)
Comment (rohithkrn) [reply rocm-docs/caffe2-build.md]: Done.

File: rocm-docs/caffe2-build.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+	``` 
+	export LD_LIBRARY_PATH=/usr/local/caffe2/lib:$LD_LIBRARY_PATH
+	export PYTHONPATH=/usr/local/caffe2/lib/python2.7/dist-packages:$PYTHONPATH
+	```
+
+	Run the binaries under `/pytorch/build_caffe2/bin`
+
+### Run benchmarks
+
+Navigate to build directory, `cd /pytorch/build_caffe2` to run benchmarks.
```
Comment (dagamayank) [inline rocm-docs/caffe2-build.md]: Do these work on upstream C2 by default or do we need port them to AMD?

File: rocm-docs/caffe2-build.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+Install ROCm stack following steps at [link](https://github.com/RadeonOpenCompute/ROCm/blob/master/README.md) if your machine doesn't have ROCm already.
+
+## Build caffe2 inside docker
+
+ If your machine doesn't have docker installed, follow the steps [here](https://docs.docker.com/install/linux/docker-ce/ubuntu/#install-docker-ce) to install docker.
+
+### Pull the docker image
+```
+docker pull rocm/caffe2:unbuilt-rocm1.8.2
```
Comment (dagamayank) [inline rocm-docs/caffe2-build.md]: Can we also provide a docker image which has Caffe2 built and installed on ROCm 1.8.2 for the user who merely wants to use C2? We can host it [here](https://hub.docker.com/r/rocm/)

File: rocm-docs/caffe2-build.md (hunk 5 above)
Comment (rohithkrn) [reply rocm-docs/caffe2-build.md]: They currently will work on rocm fork once this PR gets merged. We can make it work on upstream pytorch repo with changes to the Python API. I created a PR with those changes, but they do not want to make changes on python side until c++ runtime becomes stable.

File: rocm-docs/caffe2-build.md (hunk 6 above)
Comment (rohithkrn) [reply rocm-docs/caffe2-build.md]: Yes. I will add the docker with caffe2 installed.

**Comment by dagamayank** [comment]
Do we need a separate `quick-start guide`? What is the need for it?

File: rocm-docs/caffe2-build.md [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+cd pytorch
+```
+
+### Launch the docker container
+```	
+docker run -it --network=host --device=/dev/kfd --device=/dev/dri --group-add video -v $PWD:/pytorch rocm/caffe2:unbuilt-rocm1.8.2
+``` 
+Navigate to pytorch directory `cd /pytorch` inside the container.
+
+### Build caffe2 Project from Src
```
Comment (dagamayank) [inline rocm-docs/caffe2-build.md]: Src --> source

File: rocm-docs/caffe2-build.md [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+To run benchmarks, skip directly to benchmarks section of the document.
+
+## Build Caffe2 from source
+### Pull the docker image
+```
+docker pull rocm/caffe2:unbuilt-rocm1.8.2
+```
+This docker image has all the dependencies for caffe2 pre-installed.
+
+### Pull the latest caffe2 source:
```
Comment (dagamayank) [inline rocm-docs/caffe2-build.md]: source must be pulled outside of launching the docker? 
what all is in this container - `unbuilt-rocm1.8.2`? just bare-bones ROCm or anything related to C2 as well?

Comment (rohithkrn) [reply rocm-docs/caffe2-build.md]: Yes, just bare-bones ROCm

**Comment by rohithkrn** [comment]
Separate `quick-start guide` was mainly focused towards CQE, to point them to run tests.

File: rocm-docs/caffe2-build.md (hunk 8 above)
Comment (dagamayank) [reply rocm-docs/caffe2-build.md]: Then why is the `git clone` of C2 source outside of the docker?

Comment (rohithkrn) [reply rocm-docs/caffe2-build.md]: We clone it outside and map that directory to a directory inside docker. We can clone it directly inside the container as well.

**Comment by rohithkrn** [comment]
@dagamayank Removed quickstart guide as this doesn't seem to be the right place for it.

File: caffe2/python/convnet_benchmarks.py [hunk 9]
```diff
@@ -697,10 +776,11 @@ def GetArgumentParser():
         action='store_true',
         help="If True, dump the model prototxts to disk."
     )
-    parser.add_argument("--net_type", type=str, default="dag")
+    parser.add_argument("--net_type", type=str, default="simple")
```
Comment (ashishfarmer) [inline caffe2/python/convnet_benchmarks.py]: Any specific reason to run the Resnet model as simple net rather than default DAG?

Comment (rohithkrn) [reply caffe2/python/convnet_benchmarks.py]: I remember that we were facing some problems with dag net when layerwise_benchmarks was on. Let me check if that's not a problem anymore.

**Pull request merged**